### DISCUSSION TITLE
how  to get unread email from child inbox using exchangelib ?

### DISCUSSION THREAD

**Discussion by jxccc2009** [ecederstrand/exchangelib#1085] [Q&A]
Hi EveryOne,

how I can get unread email from child inbox using exchangelib ? 
for example ,  get unread email from todos child inbox?

```python
root
├── inbox
|   └── todos
└── archive
    ├── Last Job
    ├── GitLab issues
    └── Mom
```

thanks

**Comment by ecederstrand** [comment]
To navigate to a specific folder, you can use the slash notation:

```python
todo_folder = account.inbox / "todos"
```

To select only unread emails, you can add a filter to the query: `some_folder.filter(is_read=False)`

**Reply by jxccc2009** [reply]
It works, thanks very much

**Reply by jxccc2009** [reply]
by the way, how can I find its comprehensive Properties & methods ,; as far as I know, i got attachments,conversation_id ,last_modified_time ,datetime_sent, sender , thanks

**Reply by ecederstrand** [reply]
Technical documentation is available at https://ecederstrand.github.io/exchangelib/exchangelib/. https://ecederstrand.github.io/exchangelib/#creating-updating-deleting-sending-moving-archiving-marking-as-junk also shows how to print available fields for a given item type.

**Reply by jxccc2009** [reply]
Good morning, 
I found the properties : unique_body , only get the top body with html format, could you pls guide me how to get the whole message body with html format? using unique_body  or text_body？
thanks very much

**Reply by ecederstrand** [reply]
the `body` field is probably what you want.

**Reply by jxccc2009** [reply]
good morning , appricate your patience guideness, could you pls help me how to download the whole email to local disk, thanks.

**Reply by ecederstrand** [reply]
That depends on what you want to store, and in which format.

**Reply by jxccc2009** [reply]
i would like to save the whole mail to local disk as outlook mail format(.msg), Is it feasible? thanks

**Reply by ecederstrand** [reply]
exchangelib doesn’t support that out of the box so you’ll have to build a solution on your own. I’m not familiar with the .msg format but hopefully there’s a spec somewhere. It could just be the MIME data which is available in the `mime_content` field.
